api/src/models/user.rs

```diff
@@ -29,6 +32,28 @@ impl User {
     ) -> dsl::Filter<users::table, WithUsername<'_>> {
         users::table.filter(Self::with_username(username))
     }
+
+    /// Build a query to grant each of the given roles to the user.
+    pub fn add_roles_x(
```

Comment:
I made this `_x` to indicate that this actually executes the query, since our model methods usually just build a query and return it. I couldn't figure out the diesel types to return the query so i figured we'd just take in the connection here and execute it.

Reply:
I might be dumb but is this not async?